test/versioned/grpc/example.proto

```diff
@@ -4,17 +4,28 @@ package helloworld;
 
 // The greeting service definition.
 service Greeter {
-  // Sends a greeting
+  // unary calls
   rpc SayHello (HelloRequest) returns (HelloReply) {}
   rpc SayError (HelloRequest) returns (HelloReply) {}
+  // client side streaming calls
+  rpc SayHelloClientStream (stream HelloRequest) returns (HelloReply) {}
+  rpc SayErrorClientStream (stream HelloRequest) returns (HelloReply) {}
+  // server side streaming calls
+  rpc SayHelloServerStream (ArrayHelloRequest) returns (stream HelloReply) {}
+  rpc SayErrorServerStream (ArrayHelloRequest) returns (stream HelloReply) {}
+  // bidirectional streaming calls
+  rpc SayHelloBidiStream (stream HelloRequest) returns (stream HelloReply) {}
+  rpc SayErrorBidiStream (stream HelloRequest) returns (stream HelloReply) {}
 }
 
-// The request message containing the user's name.
 message HelloRequest {
   string name = 1;
 }
 
-// The response message containing the greetings
+message ArrayHelloRequest {
+  repeated string name = 1;
```

Comment:
Really nitpicky here, but is this the usual way to write gRPC APIs? Singular `name` even though it really is more like an array of names? It makes the rest of the code seem a bit awkward when we're iterating over a singular `name` property.

Reply:
yes this is a nuance of grpc.  it's an array that contains name fields, or repeated string of name